### PR TITLE
Use atomic instead of mutex + condvar in ParallelReadBuffer

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Use atomic instead of mutex + condvar in ParallelReadBuffer
cc: @tavplubix, @antonio2368

### CODE REVIEW

File: src/IO/ParallelReadBuffer.cpp [hunk 1]
```diff
@@ -265,8 +261,12 @@ void ParallelReadBuffer::finishAndWait()
 {
     emergency_stop = true;
 
-    std::unique_lock lock{mutex};
-    readers_done.wait(lock, [&] { return active_working_reader == 0; });
+    size_t active_readers = active_working_reader.load();
+    while (active_readers != 0)
+    {
+        active_working_reader.wait(active_readers);
```
Comment (antonio2368) [inline src/IO/ParallelReadBuffer.cpp:267]: It's good to know that atomics have `wait` function. 👌 
Is there in correctness reason for this change?

Comment (Avogar) [reply src/IO/ParallelReadBuffer.cpp:267]: Correctness was ok.

File: src/IO/ParallelReadBuffer.cpp [hunk 2]
```diff
@@ -205,12 +205,8 @@ bool ParallelReadBuffer::nextImpl()
 void ParallelReadBuffer::readerThreadFunction(ReadWorkerPtr read_worker)
 {
     SCOPE_EXIT({
-        std::lock_guard lock{mutex};
-        --active_working_reader;
-        if (active_working_reader == 0)
-        {
-            readers_done.notify_all();
-        }
+        if (--active_working_reader == 0)
```
Comment (tavplubix) [inline src/IO/ParallelReadBuffer.cpp:208]:
```suggestion
        if (active_working_reader.fetch_sub(1) == 1)
```

**Comment by tavplubix** [comment]
> Changelog entry (**a user-readable short description of the changes** that goes to CHANGELOG.md):
> Use atomic instead of mutex + condvar in ParallelReadBuffer

This description has no meaning for a user. Correct category is "Not for changelog", because this change does not affect any user-visible behavior.

**Review by tavplubix — APPROVED** [review]

**Review by antonio2368 — APPROVED** [review]

**Comment by Avogar** [comment]
@Mergifyio update

**Pull request merged**